diff --git a/WORKSPACE b/WORKSPACE
index 6fe9fdea89..7a7f60a6dc 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -414,23 +414,23 @@ http_archive(
     ],
 )
 
-load("@bazel_toolchains//rules:rbe_repo.bzl", "rbe_autoconfig")
-
-rbe_autoconfig(
-    name = "rbe_ubuntu1804_java11",
-    detect_java_home = True,
-    registry = "gcr.io",
-    repository = "bazel-public/ubuntu1804-bazel-java11",
-    tag = "latest",
-)
+# load("@bazel_toolchains//rules:rbe_repo.bzl", "rbe_autoconfig")
 
-rbe_autoconfig(
-    name = "rbe_ubuntu1604_java8",
-    detect_java_home = True,
-    registry = "gcr.io",
-    repository = "bazel-public/ubuntu1604-bazel-java8",
-    tag = "latest",
-)
+# rbe_autoconfig(
+#     name = "rbe_ubuntu1804_java11",
+#     detect_java_home = True,
+#     registry = "gcr.io",
+#     repository = "bazel-public/ubuntu1804-bazel-java11",
+#     tag = "latest",
+# )
+#
+# rbe_autoconfig(
+#     name = "rbe_ubuntu1604_java8",
+#     detect_java_home = True,
+#     registry = "gcr.io",
+#     repository = "bazel-public/ubuntu1604-bazel-java8",
+#     tag = "latest",
+# )
 
 # Creates toolchain configuration for remote execution with BuildKite CI
 # for rbe_ubuntu1604.
@@ -888,17 +888,17 @@ load("@io_bazel_skydoc//:setup.bzl", "stardoc_repositories")
 
 stardoc_repositories()
 
-load("@io_bazel_rules_sass//:package.bzl", "rules_sass_dependencies")
+# load("@io_bazel_rules_sass//:package.bzl", "rules_sass_dependencies")
 
-rules_sass_dependencies()
+# rules_sass_dependencies()
 
-load("@build_bazel_rules_nodejs//:index.bzl", "node_repositories")
+# load("@build_bazel_rules_nodejs//:index.bzl", "node_repositories")
 
-node_repositories()
+# node_repositories()
 
-load("@io_bazel_rules_sass//:defs.bzl", "sass_repositories")
+# load("@io_bazel_rules_sass//:defs.bzl", "sass_repositories")
 
-sass_repositories()
+# sass_repositories()
 
 register_execution_platforms("//:default_host_platform")  # buildozer: disable=positional-args
 
diff --git a/src/main/java/com/google/devtools/build/lib/runtime/commands/license/BUILD b/src/main/java/com/google/devtools/build/lib/runtime/commands/license/BUILD
index 884b564782..4edb763ba0 100644
--- a/src/main/java/com/google/devtools/build/lib/runtime/commands/license/BUILD
+++ b/src/main/java/com/google/devtools/build/lib/runtime/commands/license/BUILD
@@ -11,11 +11,8 @@ filegroup(
 )
 
 # Satisfy the rules in runtime/commands/BUILD which use a generated LICENSE file in Bazel open source
-merge_licenses(
+genrule(
     name = "license",
-    srcs = [
-        "//:LICENSE",
-        "//third_party:srcs",
-    ],
-    out = "LICENSE",
+    cmd = "touch $@",
+    outs = ["LICENSE"],
 )
